### DISCUSSION TITLE
How to add metadat Component when generating the SBOM?

### DISCUSSION THREAD

**Discussion by MohammedAziz02** [CycloneDX/cyclonedx-python#801] [Q&A] [answered]
Hi, I tried to generate the SBOM using this tool, I discovred that the metadata Component is missed, https://cyclonedx.org/docs/1.6/json/#metadata_component, please is there is any way I can add it ?

**Comment by jkowalleck** [comment, marked as answer]
how did you call the tool? see https://cyclonedx-bom-tool.readthedocs.io/en/latest/usage.html
depending on the method, the metadata component is added automatically, or may be added via `--pyproject` argument.

**Reply by MohammedAziz02** [reply]
I call this command : "cyclonedx-py requirements requirements.txt --of JSON --sv 1.4 -o cyclonedxPy.json", but it's seems that when I call it on the requirements file only, I don't get the metadata component ? should I use another command ?

**Reply by jkowalleck** [reply]
> when I call it on the requirements file only, I don't get the metadata component

thats why i mentioned `--pyproject`.
please read the manual: https://cyclonedx-bom-tool.readthedocs.io/en/latest/usage.html
